test/server/test_server.py

```diff
@@ -0,0 +1,106 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import time
+import threading
+import mock
+import asyncio
+
+from unittest.mock import MagicMock
+
+from src.server.server import start_server
+from src.server.server import EvaServer
+
+
+from concurrent.futures import CancelledError
+
+
+class ServerTests(unittest.TestCase):
+
+    def setUp(self):
+        self.loop = asyncio.new_event_loop()
+        self.stop_server_future = self.loop.create_future()
+        asyncio.set_event_loop(None)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def test_server(self):
+
+        host = "0.0.0.0"
+        port = 5489
+
+        def timeout_server():
+            # need a more robust mechanism for when to cancel the future
+            time.sleep(2)
+            self.stop_server_future.cancel()
+
+        thread = threading.Thread(target=timeout_server)
+        thread.daemon = True
+        thread.start()
+
+        with self.assertRaises(CancelledError):
+            start_server(host=host, port=port,
+                         loop=self.loop,
+                         stop_server_future=self.stop_server_future)
+
+    def test_server_protocol(self):
+
+        eva_server = EvaServer()
+        eva_server.transport = mock.Mock()
+        eva_server.transport.close = MagicMock(return_value="closed")
+        eva_server.transport.abort = MagicMock(return_value="aborted")
+
+        # connection made
```

Comment:
I feel this test should be split into multiple tests.

Reply:
Done.